### PR TITLE
Start process to improve Jira logging

### PR DESCRIPTION
**Pull request by grahamalama**
Related #645 

Adds / modifies logging for a few Jira service methods

- get_issue
- create_issue
- update_issue_resolution

The important things we want to check here:
- things are logged at the appropriate level (Debug)
- we include the `rid` everywhere, as this is what will help us correlate logs for an action workflow
- We test this in a sane way

This PR is getting a temperature-check for the logging / testing patterns before I continue further

### CODE REVIEW

File: jbi/services/jira.py [hunk 1]
```diff
@@ -253,33 +261,49 @@ def create_jira_issue(
...
+                message = ", ".join(str(exc).split("\n"))
+                raise JiraCreateError(message) from exc
+            except json.JSONDecodeError:
+                # If the response is not JSON, use the response text
+                logger.exception(
+                    "Failed to create issue for Bug %s",
+                    bug.id,
+                    extra={"response": exc.response.text, **context.model_dump()},
+                )
+                raise JiraCreateError(str(exc)) from exc
```
Comment (leplatrem) [inline jbi/services/jira.py:295]: Wouldn't this be more concise? Is the `join`/`split` specific to the http exception?

```suggestion
            try:
                response = exc.response.json()
            except json.JSONDecodeError:
                # If the response is not JSON, use the response text
                response = exc.response.text
            
            logger.exception(
                "Failed to create issue for Bug %s",
                bug.id,
                extra={"response": response, **context.model_dump()},
            )
            message = ", ".join(str(exc).split("\n"))
            raise JiraCreateError(message) from exc
```

File: tests/unit/services/test_jira.py [hunk 2]
```diff
@@ -155,3 +154,231 @@ def test_all_projects_components_exist_no_components_param(
...
+
+    before, after = capturelogs.records
+    assert before.levelno == logging.DEBUG
+    assert before.message == "Getting issue JBI-234"
+
+    assert after.levelno == logging.ERROR
+    assert after.message.startswith("Could not read issue JBI-234")
+
+
+def test_get_issue_raises_other_error(
```
Comment (leplatrem) [inline tests/unit/services/test_jira.py:199]: nit: other error*s* (?)


Below you have just `test_update_..._raises(` which is fine too :)

File: tests/unit/services/test_jira.py [hunk 3]
```diff
@@ -155,3 +154,231 @@ def test_all_projects_components_exist_no_components_param(
...
+
+    before, after = capturelogs.records
+    assert before.message == f"Creating new Jira issue for Bug {context.bug.id}"
+    assert before.fields == issue_fields
+
+    assert after.message == f"Jira issue JBI-234 created for Bug {context.bug.id}"
+    assert after.response == mocked_response_data
+
+
+def test_create_jira_issue_returns_list(
```
Comment (leplatrem) [inline tests/unit/services/test_jira.py:308]:
```suggestion
def test_create_jira_issue_when_list_is_returned(
```
?

**Review by leplatrem — APPROVED** [review]
👌 
(you can ignore any of my comments if they don't make sense to you)

File: jbi/services/jira.py (hunk 1 above)
Comment (grahamalama) [reply jbi/services/jira.py]: Good call. I changed things up a big more in [e7aa369](https://github.com/mozilla/jira-bugzilla-integration/pull/767/commits/e7aa36959f50b8d17602e9fa42f0c1e3759bee72) -- lmk what you think

File: jbi/services/jira.py [hunk 4]
```diff
@@ -253,33 +261,49 @@ def create_jira_issue(
...
-        jira_response_create = self.client.create_issue(fields=fields)
+        logger.debug(
+            "Creating new Jira issue for Bug %s",
+            bug.id,
+            extra={"fields": fields, **context.model_dump()},
+        )
+        try:
+            response = self.client.create_issue(fields=fields)
+        except requests.HTTPError as exc:
+            assert exc.response is not None
```
[the remark this answers is not in the document]
Comment (grahamalama) [reply jbi/services/jira.py:278]: True. This was just to make mypy happy -- we can be reasonably sure we'll have a response object here, no?

If we continue to fight with `mypy` about safely accessing `exc.response`, maybe it deserves a helper 🤔

File: tests/unit/services/test_jira.py (hunk 2 above)
Comment (grahamalama) [reply tests/unit/services/test_jira.py:199]: `nit: other errors (?)`

Maybe "raises" is the wrong word here -- my goal was to illustrate that we the Jira Client calls raise_for_status, we swallow `404`s, but reraise everything else. I'll change the name of this test to "test_get_issue_reraises_other_erroring_status_codes"

**Pull request merged**